### ISSUE TITLE
Generated body from Swagger-UI seems wrong

### ISSUE THREAD

**Issue by florian-besser** [swagger-api/swagger-ui#1645]
I'm using the current Master of swagger-ui.

This is part of the generated swagger.json:

```
{  
   "method":"PUT",
   "summary":"...",
   "notes":"",
   "type":"void",
   "nickname":"...",
   "consumes":[  
      "application/json"
   ],
   "parameters":[  
      {  
         "name":"accountId",
         "description":"accountId",
         "required":true,
         "type":"string",
         "paramType":"path",
         "allowMultiple":false
      },
      {  
         "name":"userId",
         "description":"userId",
         "required":true,
         "type":"string",
         "paramType":"path",
         "allowMultiple":false
      },
      {  
         "name":"body",
         "description":"...",
         "required":true,
         "type":"array",
         "items":{  
            "$ref":"Schedule"
         },
         "paramType":"body",
         "allowMultiple":false
      }
   ]
}
```

This is how Swagger UI renders (which is correct):

![capture1](https://cloud.githubusercontent.com/assets/7090706/10246472/0754d774-6912-11e5-8f21-00555fbc791c.PNG)

When I click on "Try Out" this happens:

![capture2](https://cloud.githubusercontent.com/assets/7090706/10246490/1f821ab4-6912-11e5-9913-c3c1f1495021.PNG)

In other words: The individual parts of the body are not transferred as one JSON String (as I would like) but instead are mangled into an array of small strings.

this means the receiving server will complain that 

```
Can not instantiate value of type [simple type, class ...Schedule] from String value ('['); no single-String constructor/factory method
```

How can I fix it so that Swagger-UI sends the equivalent of using the postman "raw" section?
![capture3](https://cloud.githubusercontent.com/assets/7090706/10246568/c70ca740-6912-11e5-9d65-023ec6a55975.PNG)

**Comment by webron** [comment]
Thanks for reporting, this is indeed a nasty one.

I assume this is because the type is array, and we have a feature that detects an array input as each value of the array per line, but this breaks in the case of full json structures. If you flatten the object to a single line, and remove the encompassing array, it should work for now, but we definitely need to resolve this.

**Comment by rwinscot** [comment]
Strange results even with flattening... something like this:

  [{"id":null,"description":"","name":"some name"}]

Turns into this...

  ["[{\"id\":null,\"description\":\"\",\"name\":\"some name\"}]"]

2.0.24 works great... master... not so much - something is happening to the body in getInputMap.

**Comment by rwinscot** [comment]
Might be an over-simplification... but by the time we arrive in Operation.prototype.getBody() ( line ~4240 in swagger-ui.js ) we can join the array and all is well.  

if (param.in === 'body') {
   body = args[param.name].join('');
} ...

Thoughts?

**Comment by fehguy** [comment]
OK this is a 1.2 spec, and swagger-js is converting them into 2.0 specs under the hood.

It appears that for `body`, we're not creating the `schema` attribute correctly, which contains `"type": "array"` inside it.

**Comment by jceloi** [comment]
Hi,

I've encountered the same problem, see #1643 referenced in this issue.

My analysis was that getTextAreaValue method was responsible for this behavior, which processes all arrays the same way, even if they are a full-fledged json object (using the content-type as a reference).

The simplistic hack works, but I did not dive deeper and I might be missing something.

**Comment by rwinscot** [comment]
Looking for comments on a potential fix - @fehguy what do you think?

// add to doc.js - to help identify what spec we should be using
var swaggerVersion = '0.0';


// add to SwaggerClient.prototype.build in swagger-client.js - sets the API version for above
swaggerVersion = self.swaggerVersion = responseObj.swaggerVersion;
self.swaggerObject = responseObj;

   if (responseObj.swagger && parseInt(responseObj.swagger) === 2) {
      swaggerVersion = self.swaggerVersion = responseObj.swagger;
      ...


// Inner block where param.in === 'body' collapses arrays for legacy APIs
Operation.prototype.getBody = function (headers, args, opts) {
  var formParams = {}, body, key, value, hasBody = false;

  for (var i = 0; i < this.parameters.length; i++) {
    var param = this.parameters[i];

    if (typeof args[param.name] !== 'undefined') {
      if (param.in === 'body') {

        // body contains an array within a strigified array... collapse
        // for consumption in old spec or new.
		if(Array.isArray(args[param.name]) && swaggerVersion <= 1.2) {
			body = args[param.name].join('');
		} else {
			body = args[param.name];
		}
      } ...

**Comment by vijrahul** [comment]
Hi,

Do we have an update on this issue? When will this issue be fixed?

**Comment by JSurf** [comment]
Should someone create a pull request with the proposed fix from @rwinscot ? 
Are there any drawbacks to the proposed solution ?

**Comment by fehguy** [comment]
That's very odd.  I have tested it on the petstore and am not seeing the result.

http://petstore.swagger.io/#!/user/createUsersWithArrayInput

@JSurf Can you please look there to see what is different with your setup?

**Comment by JSurf** [comment]
You need to test against the old v1.2 spec. The v2 spec is working

https://github.com/swagger-api/swagger-ui/tree/master/test/specs/v1.2/petstore

**Comment by webron** [comment]
It's unlikely we're going to resolve this. 3.X no longer support older versions of the spec. Consider upgrading your spec definitions.
